Add recipe for nekonight theme

BrunoCiccarino opened the pull request:
### Brief summary of what the package does

Hi,

This pull request adds the nekonight theme collection to MELPA. The collection includes multiple Emacs themes, with nekonight-moon as one of the main themes.

### Direct link to the package repository

https://github.com/neko-night/emacs

### Your association with the package

I'm an mantainer

### Relevant communications with the upstream package maintainer

- The themes are located in the emacs/ directory.
- The package metadata is defined in nekonight-pkg.el, located at the root of the repository.
- The repository is hosted on GitHub at https://github.com/neko-night/emacs
- The theme is compatible with emacs 24+

### Checklist

<!-- Please confirm by replacing `[]` with `[x]`: -->

- [x] The package is released under a [GPL-Compatible Free Software License](https://www.gnu.org/licenses/license-list.en.html#GPLCompatibleLicenses)
- [x] I've read [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)
- [x] I've used the latest version of [package-lint](https://github.com/purcell/package-lint) to check for packaging issues, and addressed its feedback
- [x] My elisp byte-compiles cleanly
- [x] I've used `M-x checkdoc` to check the package's documentation strings
- [x] I've built and installed the package using the instructions in [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)

<!-- After submitting, please fix any problems the CI reports. -->

gs-101 commented:
Hello Bruno, I've seen your post on the mailing list.

# MELPA

1. Go through the suggestions on [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org).
2. Lint your file with [package-lint](https://github.com/purcell/package-lint).
3.  Run `checkdoc` command to get more suggestions

Which is just the process in the PR checklist.

Now, I copied over your file, and here are the results:

`package-lint`:

<details>

```
1:0: warning: "Version:" or "Package-Version:" header is missing. MELPA will handle this, but other archives will not.
1:0: error: Package should have a ;;; Commentary section.
1:0: error: Package should have a Homepage or URL header.
1:0: error: package.el cannot parse this buffer: Package lacks a file header
28:6: warning: Closing parens should not be wrapped onto new lines.
374:3: warning: Closing parens should not be wrapped onto new lines.
```
</details>

The main thing I'd say you are missing is the package header. [Here](https://github.com/emacscollective/no-littering/blob/74431db47edf44fc1f5f013033560f7bc4f26f24/no-littering.el#L1-L86) is an example of one. The main fields of note are `;;; Commentary:` and `;;; Code:`, the others (as in `;;;; Usage` and `;;;; Conventions` aren't strictly necessary.

Running `checkdoc` made me generate these sections, with the only error being a missing period for the docstring.

# ELPA

Sorry if this section isn't as detailed, it's just that I've never uploaded to ELPA. But for that, there's two:

- NonGNU ELPA
- GNU ELPA

For your package to be on GNU ELPA, you have to [assign your copyright to the FSF](https://www.fsf.org/licensing/assigning.html).

For NonGNU ELPA, you have this section on the [README.org](https://git.savannah.gnu.org/cgit/emacs/nongnu.git/plain/README.org) file:

<details>

## Rules for a package to be acceptable in NonGNU ELPA

### A NonGNU ELPA package must display its copyright notices and license notices

This can be solved by adding a header to your package, just like it was showed on the MELPA section. 

## Adding a package

This section seems to require maintainer approval to be followed. I'm sure that when one of the maintainers see your e-mail, they'll guide you through this. But if you want to see it, it's the last headline before `License`.
</details>

tarsius commented:
Note that your recipe is in a format used by an unrelated project (doom emacs I believe). Use our recipe format.

Instead of adding this one file to a fork of the original vim repository, use a new repository. Some people install package by cloning their git repositories and it is unfortunate if they have to populate their file system with a hundred unrelated files, just to get their hand on one elisp library.

BrunoCiccarino commented:
Hi, thanks for giving me guidance, I really don't know how to do it

gs-101 commented:
@BrunoCiccarino 

```diff
-(package! nekonight
-  :recipe (:host github :repo "BrunoCiccarino/nekonight"
-           :files ("emacs/*.el" "nekonight-pkg.el")))
+(nekonight
+ :fetcher github :repo "BrunoCiccarino/nekonight"
+ :files ("emacs/nekonight-moon-theme.el"))
```

BrunoCiccarino commented:
how i can edit this pr?

tarsius commented:
> how i can edit this pr?

Pressingly you created this pr by forking this repository and then cloning that fork, added the recipe on your local machine, pushed the result to your fork, before finally creating the pr.

So now that you have to make changes, go to your clone again, make the changes, and push again.

tarsius commented:
This package consists of multiple themes. Please follow the conventions described [here](https://github.com/melpa/melpa/pull/8892#issuecomment-1898641161) and [here](https://github.com/melpa/melpa/blob/4caa9a03cad40d24484fce27fcfcf75b1d08e814/FAQ.md#how-do-i-make-a-package-containing-multiple-themes).

(@riscy Please publish that FAQ somewhere more convenient, such as a new page on the wiki.)

tarsius commented:
The recipe file should be named `nekonight-themes` and contain:

```elisp
(nekonight-themes :fetcher github :repo "BrunoCiccarino/nekonight")
```

BrunoCiccarino commented:
> The recipe file should be named `nekonight-themes` and contain:
> 
> ```emacs-lisp
> (nekonight-themes :fetcher github :repo "BrunoCiccarino/nekonight")
> ```

I changed the emacs theme repository as you suggested, how can I edit my pull request to include the new repository?

tarsius commented:
> I changed the emacs theme repository as you suggested, how can I edit my pull request to include the new repository?

I explained that here: https://github.com/melpa/melpa/pull/9291#issuecomment-2558265794.

tarsius commented:
> But you should probably rename the repository from `emacs` to `nekonight-themes` and then adjust that accordingly in the recipe.

Nevermind, I realized you created a new "organization" just for this repository. Well, that is highly unusual for such a young project with only one contributor, but in that case it might make sense to stick to naming the repository just `emacs` (especially if you intend to create other repositories to implement the same theme for other editors).

BrunoCiccarino commented:
> > But you should probably rename the repository from `emacs` to `nekonight-themes` and then adjust that accordingly in the recipe.
> 
> Nevermind, I realized you created a new "organization" just for this theme. Well, that is highly unusual for such a young project with only one contributor, but in that case it might make sense to stick to naming the repository just `emacs` (especially if you intend to create other repositories to implement the same theme for other editors).

Now I've renamed it 😂, that's ok... The important thing is that it works

tarsius commented:
The library `nekonight-theme.el` should be named `nekonight-themes.el` (with an S).

BrunoCiccarino commented:
> The library `nekonight-theme.el` should be named `nekonight-themes.el` (with an S).

ohhh, ok

BrunoCiccarino commented:
> The library `nekonight-theme.el` should be named `nekonight-themes.el` (with an S).

I achieved

tarsius commented:
And it should contain a proper library with all the relevant metadata. Melpa and other tools extract metadata from that file because it is the "main library" (the library whose name matches the name of the package).

[Earlier](https://github.com/melpa/melpa/pull/9291#issuecomment-2558266647) I suggested that you create this file by copying one of the other libraries and making the necessary adjustments (which are to adjust the filename mentioned in the file and maybe to adjust the summary line and commentary section to reflect the fact that this *package* consists of multiple *themes*.)

BrunoCiccarino commented:
> And it should contain a proper library with all the relevant metadata. Melpa and other tools extract metadata from that file because it is the "main library" (the library whose name matches the name of the package).
> 
> [Earlier](https://github.com/melpa/melpa/pull/9291#issuecomment-2558266647) I suggested that you create this file by copying one of the other libraries and making the necessary adjustments (which are to adjust the filename mentioned in the file and maybe to adjust the summary line and commentary section to reflect the fact that this _package_ consists of multiple _themes_.)

I copied that example and added a few more comments because I imagined that would be the description of the theme, so I added what I would like to appear

tarsius commented:
The `Package-Keywords` and `Package-Description` library headers are your inventions -- nothing recognized those.

BrunoCiccarino commented:
> The `Package-Keywords` and `Package-Description` library headers are your inventions -- nothing recognized those.

So is this what is giving you the lint error? I will take

tarsius commented:
The `Package-Keywords` and `Package-Description` library headers are your own inventions -- nothing recognized those. Use `Keywords` instead -- but I am not sure what an appropriate value for themes is -- `themes` is not one of the known keywords (which you can list with `M-x finder-list-keywords` -- might just skip that header. There also is no `Description` header. The "package summary" is taken from the first line of the file instead.

> So is this what is giving you the lint error?

No. Non-standard header keywords do not cause any lint warnings.

tarsius commented:
The error you have to address next is:

`Recipe 'nekonight' contains mismatched package name 'nekonight-themes'`

BrunoCiccarino commented:
> The `Package-Keywords` and `Package-Description` library headers are your own inventions -- nothing recognized those. Use `Keywords` instead -- but I am not sure what an appropriate value for themes is -- `themes` is not one of the known keywords (which you can list with `M-x finder-list-keywords`) -- might just skip that header. There also is no `Description` header. The "package summary" is taken from the first line of the file instead.
> 
> > So is this what is giving you the lint error?
> 
> No. Non-standard header keywords do not cause any lint warnings.

How can I resolve this lint error?

tarsius commented:
> How can I resolve this lint error?

The error you have to address next is:

`Recipe 'nekonight' contains mismatched package name 'nekonight-themes'`

You fix that by renaming the recipe file so that it matches the intended package name (`nekonight-themes`).

BrunoCiccarino commented:
Phew, I got it

tarsius commented:
> > And it should contain a proper library with all the relevant metadata. Melpa and other tools extract metadata from that file because it is the "main library" (the library whose name matches the name of the package).
> > [Earlier](https://github.com/melpa/melpa/pull/9291#issuecomment-2558266647) I suggested that you create this file by copying one of the other libraries and making the necessary adjustments (which are to adjust the filename mentioned in the file and maybe to adjust the summary line and commentary section to reflect the fact that this _package_ consists of multiple _themes_.)
> 
> I copied that example and added a few more comments because I imagined that would be the description of the theme, so I added what I would like to appear

I still suggest that you make the header of this libraries look more like the headers of the other libraries.

Once you have done that, the regular review can begin (which @riscy will do).

I'm off now.

BrunoCiccarino commented:
> > > And it should contain a proper library with all the relevant metadata. Melpa and other tools extract metadata from that file because it is the "main library" (the library whose name matches the name of the package).
> > > [Earlier](https://github.com/melpa/melpa/pull/9291#issuecomment-2558266647) I suggested that you create this file by copying one of the other libraries and making the necessary adjustments (which are to adjust the filename mentioned in the file and maybe to adjust the summary line and commentary section to reflect the fact that this _package_ consists of multiple _themes_.)
> > 
> > 
> > I copied that example and added a few more comments because I imagined that would be the description of the theme, so I added what I would like to appear
> 
> I still suggest that you make the header of this libraries look more like the headers of the other libraries.
> 
> Once you have done that, the regular review can begin (which @riscy will do).
> 
> I'm off now.

Ok, I'll do it, in the metadata, right?

tarsius commented:
> > > > And it should contain a proper library with all the relevant metadata. Melpa and other tools extract metadata from that file because it is the "main library" (the library whose name matches the name of the package).
> > > > [Earlier](https://github.com/melpa/melpa/pull/9291#issuecomment-2558266647) I suggested that you create this file by copying one of the other libraries and making the necessary adjustments (which are to adjust the filename mentioned in the file and maybe to adjust the summary line and commentary section to reflect the fact that this _package_ consists of multiple _themes_.)
> > > 
> > > 
> > > I copied that example and added a few more comments because I imagined that would be the description of the theme, so I added what I would like to appear
> > 
> > 
> > I still suggest that you make the header of this libraries look more like the headers of the other libraries.
> > Once you have done that, the regular review can begin (which @riscy will do).
> > I'm off now.
> 
> Ok, I'll do it, in the metadata, right?

What you have done in https://github.com/neko-night/emacs/commit/3047bdb1adc28162cc275ac649fffd93828c64b9 is the opposite of what I suggested you do. You should not remove the last pieces of metadata, you should add the missing pieces.

I.e., the file should look like this:

```elisp
;;; nekonight-themes.el --- Neko Night Themes -*- lexical-binding: t -*-

;; Copyright (C) 2024 Bruno Ciccarino

;; Author: Bruno Ciccarino
;; URL: https://github.com/BrunoCiccarino/nekonight
;; Keywords: faces, theme, nekonight, moon

;; Package-Version: 0.0.3
;; Package-Requires: ((emacs "24.0"))

;; SPDX-License-Identifier: MIT

;; This file is not part of GNU Emacs.

;; Permission is hereby granted, free of charge, to any person obtaining a copy
;; of this software and associated documentation files (the "Software"), to deal
;; in the Software without restriction, including without limitation the rights
;; to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
;; copies of the Software, and to permit persons to whom the Software is
;; furnished to do so, subject to the following conditions:

;; The above copyright notice and this permission notice shall be included in all
;; copies or substantial portions of the Software.

;; THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
;; IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
;; FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
;; AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
;; LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
;; OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
;; SOFTWARE.

;;; Commentary:

;; Looking for a theme that fits your vibe? Whether you're into dark mode,
;; light mode, or something in between, this NekoNight has got you covered.
;; It's a buffet of color schemes for every taste and mood.

;;; Code:

(defgroup nekonight nil
  "Neko Night themes."
  :prefix "nekonight-"
  :group 'faces)

(defvar nekonight-main-color "#1a1b26"
  "Main color used in Neko Night themes.")

;;;###autoload
(add-to-list
 'custom-theme-load-path
 (if load-file-name (file-name-directory load-file-name) default-directory))

(provide 'nekonight-themes)
;;; nekonight-themes.el ends here
```

BrunoCiccarino commented:
> > > > > E deve conter uma biblioteca adequada com todos os metadados relevantes. Melpa e outras ferramentas extraem metadados desse arquivo porque é a "biblioteca principal" (a biblioteca cujo nome corresponde ao nome do pacote). 
> > > > > [Anteriormente,](https://github.com/melpa/melpa/pull/9291#issuecomment-2558266647) sugeri que você criasse esse arquivo copiando uma das outras bibliotecas e fazendo os ajustes necessários (que são para ajustar o nome do arquivo mencionado no arquivo e talvez para ajustar a linha de resumo e a seção de comentários para refletir o fato de que esse _pacote_ consiste em vários _temas_ .)
> > > > 
> > > > 
> > > > Copiei esse exemplo e adicionei mais alguns comentários porque imaginei que seria a descrição do tema, então adicionei o que gostaria que aparecesse
> > > 
> > > 
> > > Eu ainda sugiro que você faça o cabeçalho dessas bibliotecas parecer mais com os cabeçalhos das outras bibliotecas. 
> > > Depois de fazer isso, a revisão regular pode começar (o que@riscyfarei). 
> > > Estou indo agora.
> > 
> > 
> > Ok, farei isso nos metadados, certo?
> 
> O que você fez em [neko-night/emacs@ 3047bdb](https://github.com/neko-night/emacs/commit/3047bdb1adc28162cc275ac649fffd93828c64b9) é o oposto do que eu sugeri que você fizesse. Você não deve remover as últimas partes de metadados, você deve adicionar as partes faltantes.
> 
> Ou seja, o arquivo deve ficar assim:
> 
> ```emacs-lisp
> ;;; nekonight-themes.el --- Neko Night Themes -*- lexical-binding: t -*-
> 
> ;; Copyright (C) 2024 Bruno Ciccarino
> 
> ;; Author: Bruno Ciccarino
> ;; URL: https://github.com/BrunoCiccarino/nekonight
> ;; Keywords: faces, theme, nekonight, moon
> 
> ;; Package-Version: 0.0.3
> ;; Package-Requires: ((emacs "24.0"))
> 
> ;; SPDX-License-Identifier: MIT
> 
> ;; This file is not part of GNU Emacs.
> 
> ;; Permission is hereby granted, free of charge, to any person obtaining a copy
> ;; of this software and associated documentation files (the "Software"), to deal
> ;; in the Software without restriction, including without limitation the rights
> ;; to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
> ;; copies of the Software, and to permit persons to whom the Software is
> ;; furnished to do so, subject to the following conditions:
> 
> ;; The above copyright notice and this permission notice shall be included in all
> ;; copies or substantial portions of the Software.
> 
> ;; THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
> ;; IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
> ;; FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
> ;; AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
> ;; LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
> ;; OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
> ;; SOFTWARE.
> 
> ;;; Commentary:
> 
> ;; Looking for a theme that fits your vibe? Whether you're into dark mode,
> ;; light mode, or something in between, this NekoNight has got you covered.
> ;; It's a buffet of color schemes for every taste and mood.
> 
> ;;; Code:
> 
> (defgroup nekonight nil
>   "Neko Night themes."
>   :prefix "nekonight-"
>   :group 'faces)
> 
> (defvar nekonight-main-color "#1a1b26"
>   "Main color used in Neko Night themes.")
> 
> ;;;###autoload
> (add-to-list
>  'custom-theme-load-path
>  (if load-file-name (file-name-directory load-file-name) default-directory))
> 
> (provide 'nekonight-themes)
> ;;; nekonight-themes.el ends here
> ```

now it's clearer, thanks for the help

riscy commented:
> @riscy Please publish that FAQ somewhere more convenient, such as a new page on the wiki.

@tarsius This was a needed reminder.  I've let that thing flounder for *way* too long.  I'll find a home for it. :)

@BrunoCiccarino, here's a quick first pass.  Tackle as much as you can and I'll be back next weekend -

⸺ `nekonight-themes.el` with package-lint 20241127.1826 and `package-lint-main-file` = "nekonight-themes.el":
```
2 issues found:
41:0: error: "nekonight" doesn't start with package's prefix "nekonight-themes".
46:0: error: "nekonight-main-color" doesn't start with package's prefix "nekonight-themes".
```

These namespacing/prefix issues can be solved by following the aforementioned FAQ.

⸺ `nekonight-storm-theme.el` with package-lint 20241127.1826 and `package-lint-main-file` = "nekonight-themes.el":
```
5 issues found:
1:0: error: There is no `provide' form.
10:0: error: Package-Requires outside the main file have no effect.
42:0: error: "nekonight-storm" doesn't start with package's prefix "nekonight-themes".
70:6: warning: Closing parens should not be wrapped onto new lines.
464:3: warning: Closing parens should not be wrapped onto new lines.
```

I think the `provide` issue is a false positive and you can disregard.

⸺ `nekonight-moon-theme.el` with package-lint 20241127.1826 and `package-lint-main-file` = "nekonight-themes.el":
```
5 issues found:
1:0: error: "nekonight-moon" doesn't start with package's prefix "nekonight-themes".
1:0: error: There is no `provide' form.
29:6: warning: Closing parens should not be wrapped onto new lines.
81:1: warning: Closing parens should not be wrapped onto new lines.
423:3: warning: Closing parens should not be wrapped onto new lines.
```

⸺ `nekonight-storm-theme.el` with checkdoc 0.6.2 (fix *within reason*):
```
nekonight-storm-theme.el:43: First sentence should end with punctuation
```

⸺ `nekonight-moon-theme.el` with checkdoc 0.6.2 (fix *within reason*):
```
nekonight-moon-theme.el:0: The first line should be of the form: ";;; package --- Summary"
nekonight-moon-theme.el:0: You should have a section marked ";;; Commentary:"
nekonight-moon-theme.el:1: You should have a section marked ";;; Code:"
nekonight-moon-theme.el:2: First sentence should end with punctuation
```

`#'load`-check on each file:
```
Loading nekonight-themes.el
Loading nekonight-storm-theme.el
Loading nekonight-moon-theme.el
  nekonight-moon-theme.el:Error: Emacs 29.4:
  (invalid-read-syntax ")" 423 5)
```
The last line here makes me think there's an extra parenthesis somewhere in that last file.

⸺ Package and license:
- themes/nekonight-moon-theme.el needs *formal* license boilerplate and/or an [SPDX-License-Identifier](https://spdx.dev/ids/)
- themes/nekonight-moon-theme.el -- not in package namespace `nekonight-themes-`
- themes/nekonight-moon-theme.el -- no packaging header
- themes/nekonight-storm-theme.el -- not in package namespace `nekonight-themes-`

BrunoCiccarino commented:
> > @riscy Please publish that FAQ somewhere more convenient, such as a new page on the wiki.
> 
> @tarsius This was a needed reminder. I've let that thing flounder for _way_ too long. I'll find a home for it. :)
> 
> @BrunoCiccarino, here's a quick first pass. Tackle as much as you can and I'll be back next weekend -
> 
> ⸺ `nekonight-themes.el` with package-lint 20241127.1826 and `package-lint-main-file` = "nekonight-themes.el":
> 
> ```
> 2 issues found:
> 41:0: error: "nekonight" doesn't start with package's prefix "nekonight-themes".
> 46:0: error: "nekonight-main-color" doesn't start with package's prefix "nekonight-themes".
> ```
> 
> These namespacing/prefix issues can be solved by following the aforementioned FAQ.
> 
> ⸺ `nekonight-storm-theme.el` with package-lint 20241127.1826 and `package-lint-main-file` = "nekonight-themes.el":
> 
> ```
> 5 issues found:
> 1:0: error: There is no `provide' form.
> 10:0: error: Package-Requires outside the main file have no effect.
> 42:0: error: "nekonight-storm" doesn't start with package's prefix "nekonight-themes".
> 70:6: warning: Closing parens should not be wrapped onto new lines.
> 464:3: warning: Closing parens should not be wrapped onto new lines.
> ```
> 
> I think the `provide` issue is a false positive and you can disregard.
> 
> ⸺ `nekonight-moon-theme.el` with package-lint 20241127.1826 and `package-lint-main-file` = "nekonight-themes.el":
> 
> ```
> 5 issues found:
> 1:0: error: "nekonight-moon" doesn't start with package's prefix "nekonight-themes".
> 1:0: error: There is no `provide' form.
> 29:6: warning: Closing parens should not be wrapped onto new lines.
> 81:1: warning: Closing parens should not be wrapped onto new lines.
> 423:3: warning: Closing parens should not be wrapped onto new lines.
> ```
> 
> ⸺ `nekonight-storm-theme.el` with checkdoc 0.6.2 (fix _within reason_):
> 
> ```
> nekonight-storm-theme.el:43: First sentence should end with punctuation
> ```
> 
> ⸺ `nekonight-moon-theme.el` with checkdoc 0.6.2 (fix _within reason_):
> 
> ```
> nekonight-moon-theme.el:0: The first line should be of the form: ";;; package --- Summary"
> nekonight-moon-theme.el:0: You should have a section marked ";;; Commentary:"
> nekonight-moon-theme.el:1: You should have a section marked ";;; Code:"
> nekonight-moon-theme.el:2: First sentence should end with punctuation
> ```
> 
> `#'load`-check on each file:
> 
> ```
> Loading nekonight-themes.el
> Loading nekonight-storm-theme.el
> Loading nekonight-moon-theme.el
>   nekonight-moon-theme.el:Error: Emacs 29.4:
>   (invalid-read-syntax ")" 423 5)
> ```
> 
> The last line here makes me think there's an extra parenthesis somewhere in that last file.
> 
> ⸺ Package and license:
> 
> * themes/nekonight-moon-theme.el needs _formal_ license boilerplate and/or an [SPDX-License-Identifier](https://spdx.dev/ids/)
> * themes/nekonight-moon-theme.el -- not in package namespace `nekonight-themes-`
> * themes/nekonight-moon-theme.el -- no packaging header
> * themes/nekonight-storm-theme.el -- not in package namespace `nekonight-themes-`

I've already added spdx, and regarding the syntax error, I've already fixed it, but the background color is bugged at the moment, I'll try to fix lua neko, if it doesn't work I'll remove it from the package
